src/python_testing/TC_RR_1_1.py

```diff
@@ -345,6 +366,25 @@ async def test_TC_RR_1_1(self):
             if (num_failed_clients > 0):
                 asserts.fail(f"Failed Step 8b: ({num_failed_clients} / {len(client_list)} failed)")
 
+        if check_heap_watermarks:
+            # Step 8c: Read heap watermarks after
+            logging.info("Step 8c: Read Heap info after")
+
+            large_read_contents = [
+                Clusters.SoftwareDiagnostics.Attributes.CurrentHeapHighWatermark,
+                Clusters.SoftwareDiagnostics.Attributes.CurrentHeapUsed,
+            ]
+            large_read_paths = [(0, attrib) for attrib in large_read_contents]
+            swdiag_info = await dev_ctrl.ReadAttribute(self.dut_node_id, large_read_paths)
+
+            # Make sure everything came back from the read that we expected
+            asserts.assert_true(0 in swdiag_info.keys(), "Must have read endpoint 0 data")
+            asserts.assert_true(Clusters.SoftwareDiagnostics in swdiag_info[0].keys(
+            ), "Must have read Software Diagnostics cluster data")
+            for attribute in large_read_contents:
+                asserts.assert_true(attribute in swdiag_info[0][Clusters.SoftwareDiagnostics],
+                                    "Must have read back attribute %s" % (attribute.__name__))
```

Comment:
Same comment as before about names of the `large_read_*`.

Also, please log a user-friendly basic report of the before, after and delta between before and after.

If stack watermarks can be obtained, it would be good as well.